### PR TITLE
Expose URI that was deeplinked through a parameter

### PR DESCRIPTION
**Pull request by MichaelEvans**

### CODE REVIEW

**Comment by MichaelEvans** [comment]
Addresses #7

**Comment by felipecsl** [comment]
I'd like to see a test around this change to guarantee that the generated intent has this new key in the bundle. Please let me know if you need directions on how to do it

File: deeplinkdispatch-processor/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkProcessor.java [hunk 1]
```diff
@@ -187,6 +187,7 @@ private void generateDeepLinkActivity() throws IOException {
         .endControlFlow()
         .addStatement("parameterMap.put(queryParameter, uri.getQueryParameter(queryParameter))")
         .endControlFlow()
+        .addStatement("parameterMap.put(DeepLink.URI, hostPath)")
```
Comment (cdeonier) [inline deeplinkdispatch-processor/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkProcessor.java]: We should have an intact URI with the scheme, so we should replace `hostPath` with `uri.toString()`.

**Comment by MichaelEvans** [comment]
@felipecsl Yeah, if you could give me some pointers on how I could test this, I'd gladly write some or it. I wanted to submit one, but I wasn't sure how best to test that.

**Comment by felipecsl** [comment]
@MichaelEvans looks like none of the existing test are suitable to cover this test scenario. I'll set up some Robolectric tests that will make this easier. Hang tight!

**Comment by MichaelEvans** [comment]
@felipecsl added some assertions to the tests from #11 to verify the behavior

**Comment by cdeonier** [comment]
:boat:

**Comment by felipecsl** [comment]
yay! thanks @MichaelEvans

**Comment by MichaelEvans** [comment]
woot! thanks!

**Pull request merged**